### ISSUE TITLE
wx: Open/cancel an entry with multiple history entries gets a spurious change warning

### ISSUE THREAD

**Issue by nobugshere** [pwsafe/pwsafe#1010]
On macOS and Linux, when I open an entry that has multiple old passwords in the history list, then immediately click cancel, I get the warning pop-up that something has changed.  This appears to be the result of a difference in the way AddEditPropSheetDlg::PreparePasswordHistory() sorts the list from the edit dialog vs. how it's stored in the actual DB.  One ill fated attempt to correct this was in #1001.  A better way would be to re-work AddEditPropSheetDlg::PreparePasswordHistory() to match the way the DB records the list.

**Comment by nobugshere** [comment]
Proposed fix: PR #1048
Factored-out some duplicated code from three different places.  The new routine should result in the same ordering as before, but now the sorting and trimming will be more consistent.

**Comment by nobugshere** [comment]
After some more work and thought, I submitted PR #1055 as an alternative to PR #1048.  Some of the problems I encountered:
1. False mis-compare when canceling the edit dialog due to AddEditPropSheetDlg::PreparePasswordHistory sorting newest first, whereas the DB entry is nominally oldest first.  (This is the original problem.)  The easy fix would have been to just sort both lists before comparing, but wait, there's more!
2. If the max limit is ever reduced, the list would be sorted newest first, trimmed from the end, and stored newest first.
3. Next time the PW is changed, CItemData::UpdatePasswordHistory would use push_back to add to the end of the list then trim from the *front* without re-sorting, thus removing the wrong entry if the list had already been stored newest first upon reaching the limit.
4. Need to handle existing DB entries that may already be in an undefined order.

I like the new approach better because it encapsulates and standardizes the sorting and formatting of the string that is stored in the safe.  It also eliminates the duplication of the code that formats the string for storage.
